File: api/src/test/java/net/blueshell/api/config/FactoryConfig.java
package net.blueshell.api.config;

import com.github.javafaker.Faker;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Locale;
import java.util.Random;

/**
 * Test bean configuration for repeatable random data.
 */
@Configuration
public class FactoryConfig {

    @Bean
    public Faker faker() {
        return new Faker(Locale.ENGLISH);
    }

    @Bean
    public Random random() {
        return new Random();
    }
}

File: api/src/test/kotlin/net/blueshell/api/config/FactoryConfig.kt
package net.blueshell.api.platform.config

import com.github.javafaker.Faker
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.*

/**
 * Test bean configuration for repeatable random data.
 */
@Configuration
class FactoryConfig {

    @Bean
    fun faker(): Faker = Faker(Locale.ENGLISH)

    @Bean
    fun random(): Random = Random()
}
